syntax = "proto3";
package recastx.rpc;

import "google/protobuf/empty.proto";

service Control {
  rpc SetServerState (ServerState) returns (google.protobuf.Empty) {}

  rpc SetScanMode (ScanMode) returns (google.protobuf.Empty) {}
}

message ServerState {
  enum State {
    UNKNOWN = 0;
    READY = 1;
    ACQUIRING = 2;
    INITIALISING = 4;
    PROCESSING = 8;
  }

  State state = 1;
}

message ScanMode {
  enum Mode {
    CONTINUOUS = 0;
    DISCRETE = 1;
  }

  Mode mode = 1;
  uint32 update_interval = 2;
}
